functions/go/gatekeeper/README.md

```diff
@@ -4,29 +4,87 @@
 
 <!--mdtogo:Short-->
 
-Validate the KRM resources using [Gatekeeper] policies.
+Validate the KRM resources using [Gatekeeper] policies. The `gatekeeper`
+function wraps the `Gatekeeper` policy enforcement engine.
+
+[Gatekeeper] provides an extensible policy enforcement engine which allows users
+to validate the KRM resources against the Gatekeeper policies.
+
+There are 2 kinds of resources needed to define a policy:
+- [Constraint Template]: Define the schema and logic of a policy. The policy
+  logic in a Constraint Template must be written in the [Rego] language.
+- [Constraint]: Signal the Gatekeeper the corresponding constraints need
+  to be enforced. Every Constraint must be backed by a Constraint Template.
+
+To make it do something meaningful, you should to define at least one
```

Comment:
this is describing a corner case, not sure why it's in the overview. I suggest removing it.